patches/tModLoader/Terraria.ModLoader/GlobalNPC.cs

```diff
@@ -500,6 +500,31 @@ public class GlobalNPC
 		public virtual void SetupTravelShop(int[] shop, ref int nextSlot) {
 		}
 
+		/// <summary>
+		/// Wether this NPC can be telported to the King statue. Return true to allow the NPC to teleport to the statue, return false to block this NPC from teleporting to the statue, and return null to use the vanilla code for whether the NPC can teleport to the statue. Returns null by default.
+		/// </summary>
+		/// <param name="npc">The NPC</param>
+		public virtual bool? CanGoToKingStatue(NPC npc) {
+			return null;
```

Comment:
We could potentially mitigate nullability by making the default behaviour 
`return npc.modNpc?.CanGoToKingStatue() ?? vanillaTrueFalse`
which is your defaultCanGo thing
(nulls may be unnecessary evil here)